pennylane/devices/default_tensor.py

```diff
@@ -337,24 +313,39 @@ def dtype(self):
         """Tensor complex data type."""
         return self._dtype
 
-    def _reset_state(self) -> None:
+    def _reset_mps(self, wires: qml.wires.Wires) -> None:
         """
-        Reset the MPS.
+        Reset the MPS associated with the circuit.
+
+        Internally, it uses `quimb`'s `CircuitMPS` class.
 
-        This method modifies the tensor state of the device.
+        Args:
+            wires (Wires): The wires to reset the MPS.
         """
-        self._circuitMPS = qtn.CircuitMPS(psi0=self._initial_mps())
+        self._quimb_mps = qtn.CircuitMPS(
```

Comment:
Will the previous value/object stored in `_quimb_mps` be released automatically?

Reply:
This should happen automatically in Python since the garbage collector reclaims the memory. There should be no other references to the object since the `_reset_mps` method is internal and not supposed to be called from the outside. BTW, this is exactly why I don't like GC languages (I would prefer to have more control over memory management and what is allocated on the heap/stack)